tests/libvirt/models/compute/server_tests.rb

```diff
@@ -79,5 +82,45 @@
       end
       test("with q35 machine type on x86_64") { server.to_xml.match?(%r{<type arch="x86_64" machine="q35">hvm</type>}) }
     end
+    test("with efi firmware") do
+      server = Fog::Libvirt::Compute::Server.new(
+        {
+          :os_firmware => "efi",
+          :os_firmware_features => {
+            "secure-boot" => "no",
+            "enrolled-keys" => "no"
+          },
+          :nics => [],
+          :volumes => []
+        }
+      )
+      xml = server.to_xml
+
+      os_firmware = xml.include?('<os firmware="efi">')
+      secure_boot = !xml.include?('<feature name="secure-boot" enabled="no" />')
+      enrolled_keys = !xml.include?('<feature name="enrolled-keys" enabled="no" />')
```

Comment:
I think this isn't a good test because there's a very high chance the feature is just slightly different. In particular, when I run this test manually I see it includes:
```xml
    <firmware>
      <feature name="secure-boot" enabled="no"/>
      <feature name="enrolled-keys" enabled="no"/>
    </firmware>
```

Note the space before `/>` is missing.

Did you intend:
```suggestion
      secure_boot = xml.include?('<feature name="secure-boot" enabled="no"/>')
      enrolled_keys = xml.include?('<feature name="enrolled-keys" enabled="no"/>')
```